internal/pipeline/pipeline.go

```diff
@@ -137,59 +65,16 @@ func (p *Pipeline) fnChain() ([]kio.Filter, error) {
 	return runners, nil
 }
 
-// New returns a pointer to a new default Pipeline.
-// The default Pipeline should be:
-// apiVersion: kpt.dev/v1alpha1
-// kind: Pipeline
-// metadata:
-//   name: pipeline
-// sources:
-//   - './*'
-func New() *Pipeline {
-	return &Pipeline{
-		ResourceMeta: yaml.ResourceMeta{
-			TypeMeta: yaml.TypeMeta{
-				APIVersion: kptAPIVersion,
-				Kind:       pipelineKind,
-			},
-			ObjectMeta: yaml.ObjectMeta{
-				NameMeta: yaml.NameMeta{
-					Name: defaultName,
-				},
-			},
-		},
-		Sources: defaultSources,
-	}
-}
-
 // fromBytes returns a Pipeline parsed from bytes
-func fromBytes(b []byte) (*Pipeline, error) {
-	p := New()
+func fromBytes(b []byte) (*v1alpha2.Pipeline, error) {
+	p := &v1alpha2.Pipeline{}
 	err := yaml.Unmarshal(b, p)
 	if err != nil {
 		return nil, fmt.Errorf("failed to construct pipeline from bytes: %w", err)
 	}
 	return p, nil
 }
 
-// FromReader returns a Pipeline parsed from the content in reader
-func FromReader(r io.Reader) (*Pipeline, error) {
-	b, err := ioutil.ReadAll(r)
-	if err != nil {
-		return nil, fmt.Errorf("failed to construct pipeline from reader: %w", err)
-	}
-	p, err := fromBytes(b)
-	if err != nil {
-		return nil, err
-	}
-	return p, p.Validate()
-}
-
-// FromFile returns a Pipeline read from file
-func FromFile(path string) (*Pipeline, error) {
-	f, err := os.Open(path)
-	if err != nil {
-		return nil, fmt.Errorf("failed to open path %s: %w", path, err)
-	}
-	return FromReader(f)
+func fromString(s string) (*v1alpha2.Pipeline, error) {
```

Comment:
I don't see any use of these two functions, since pipeline will be read as part of reading the Kptfile, so should we remove them ?

Reply:
These 2 functions are used for testing. I will move them to test files.